packages/frontend-shared/src/graphql/urqlClient.ts

```diff
@@ -75,13 +62,16 @@ interface AppUrqlClientConfig {
 
 export type UrqlClientConfig = LaunchpadUrqlClientConfig | AppUrqlClientConfig
 
-export function makeUrqlClient (config: UrqlClientConfig): Client {
+export async function makeUrqlClient (config: UrqlClientConfig): Promise<Client> {
   let hasError = false
 
+  const dfd = pDefer()
   const exchanges: Exchange[] = [dedupExchange]
 
   const io = window.ws ?? getPubSubSource(config)
 
+  io.on('connect', dfd.resolve)
```

Comment:
Are there any errors we need to be listening to and rejecting here?

Reply:
I don't believe so the `connect_error` might be an option, but that retries, so in theory the `connect` should eventually hit. If it doesn't, it means the electron socket server has been terminated, in which the browser shouldn't even be open. Something else bigger would be wrong, and there wouldn't be a way to recover.